### PR TITLE
Add none as option for fail-on-severity

### LINKED ISSUE #431: Add the value none as the possible values for the fail-on-severity parameter
Add the value `none` as the possible values for the `fail-on-severity` parameter

## Problem

- It would be useful to provide a parameter to the action that will only list all the anomalies and do not fail
- The idea is to match one of the `none` value that can be used in the "Code Scanning" options
     -  Code Scanning alerts are still raised, but the PR is not failing/blocked
     - see [Defining the severities causing pull request check failure](https://docs.github.com/en/enterprise-cloud@latest/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/customizing-code-scanning#defining-the-severities-causing-pull-request-check-failure)

## Solution

- Add the value `none` as the possible values for the `fail-on-severity` parameter
- when set to `none` the action will:
   - not block the PR (raise an alert) in one or more vulnerabilities, or invalid licences are found
   - a warning will be publish
   - all alerts will be printed in the summary (vulnerabilities & licences)
- Default behaviour is not changed and the value is set to `low`

### PR DESCRIPTION
**Pull request by tgrall**
## This PR contains: 

- [x] Add the value none as the possible values for the fail-on-severity parameter
- [x] not block the PR (raise an alert) in one or more vulnerabilities, or invalid licences are found
- [x] a warning is publish
- [x] all alerts are  printed in the summary (vulnerabilities & licences)
- [x] Default behaviour is not changed and the value is set to low

### Checks

- [x] Document `none` behaviour
- [x] Run tests 
- [x] Tested with various PR (on my repos)

---

close #431

### CODE REVIEW

**Comment by febuiles** [comment]
@courtneycl more context in https://github.com/actions/dependency-review-action/issues/431, thoughts?

**Review by courtneycl — COMMENTED** [review]
Thanks for your contribution and your patience here, @tgrall! 

My only thought is around how it's a little inconsistent that `fail-on-severity: none` covers both vulnerabilities and licenses, when severity just references vulnerabilities. Is it clearer to have a separate parameter, like `warn-only: true`, that would make it not fail? Does that introduce other complexity or user wonkiness? @febuiles do you have any thoughts? 

Regardless, I think this is helpful and would be fine to ship as-is. I'll leave the approval for @febuiles to do a technical pass (it worked great for me when testing).

**Comment by juxtin** [comment]
Thanks again for your contribution, @tgrall! I think the basic idea here is sound, but I agree with @courtneycl about blurring the line between failing on licenses vs failing on vulnerabilities. I would feel more comfortable with keeping the two concepts separate across the possible values for `fail-on-severity`.

When it comes to licenses, we already have a lot of control for which ones are allowed. Is part of the problem you're addressing here that you'd like to still _warn_ on certain licenses, but not block the PR? If so, perhaps `warn-only: true` would be a better approach, as Courtney suggested.

Let me know what you think! I know this PR has been open for a while so I'm eager to work with you to get it merged.

**Comment by prassiv** [comment]
Hey @tgrall . I am interested in this "warn-only: true" option.I would be happy to contribute if you want me to.

**Comment by tgrall** [comment]
Hey @juxtin @courtneycl 

I have finally update my PR to create and use a new parameter `warn-only`.

I have tested the behaviour directly on workflow, but I do not know how to write a automatic test for this feature. (I have only created a test for the configuration)

Let me know if You need more do add more test and pointing me to the a good starting point, or anything else

**Comment by prassiv** [comment]
Hey @juxtin @courtneycl  @febuiles @cnagadya  -  Any feedback on this PR? I am interested in this PR and would love to help get it merged

File: __tests__/config.test.ts
Comment (juxtin) [inline __tests__/config.test.ts:30]: If this is meant to test the default, is it really necessary to have this line?

Comment (tgrall) [reply __tests__/config.test.ts:30]: Not necessary, removing it

**Comment by tgrall** [comment]
@juxtin : I have reworked the PR:
- merge the change from main
- applied your comment from July

File: README.md
Comment (jonjanego) [inline README.md:85]: should be ```vulnerabilities```

File: README.md
Comment (jonjanego) [inline README.md:91]: should be ```vulnerabilities```

File: README.md
Comment (jonjanego) [reply README.md:85]: Suggest a bit of a rewording, something like:  

> When set to ```true```, the action will log all vulnerabilities as warnings regardless of the severity, and the action will complete with a ```success``` status. This overrides the ```fail-on-severity``` option.

File: action.yml
Comment (jonjanego) [inline action.yml:65]: Suggest:

>  When set to `true` this action will always complete with `success`, overriding the `fail-on-severity` parameter.
Comment

**Review by jonjanego — CHANGES_REQUESTED** [review]
readme and wording changes

**Comment by juxtin** [comment]
Big thanks to @jonjanego for the review! I certainly endorse his comments and I'll be happy to merge this when those are addressed.

**Comment by tgrall** [comment]
@juxtin : I have applied @jonjanego suggestions

**Review by jonjanego — APPROVED** [review]
lgtm

File: action.yml
Comment (juxtin) [inline action.yml:66]: Is this line intentional?

Comment (tgrall) [reply action.yml:66]: No :( - let me work on this

**Pull request merged**
